npm/webpack-dev-server/src/CypressCTWebpackPlugin.ts

```diff
@@ -68,6 +68,16 @@ export class CypressCTWebpackPlugin {
     }
   };
 
+  private beforeCompile = () => {
+    if (!this.compilation) {
+      return
+    }
+
+    // Ensure we don't try to load files that have been removed from the file system
+    // but have not yet been detected by the onSpecsChange handler
+    this.files = (this.files || []).filter((file) => fs.existsSync(file.absolute))
```

Comment:
I am not sure we want to use `existsSync` here, should we use the async `pathExists` from `fs-extra` and `tapAsync`?

Eg: https://github.com/cypress-io/cypress/compare/tbiethman/fix/unify-1697-ct-file-removed...lmiller1990/fix/unify-1697-ct-file-removed?expand=1

Reply:
https://webpack.js.org/api/compiler-hooks/#beforecompile can work with `tapAsync`

Reply:
Or - is the point we **must** do this async, so we avoid the race condition?

Reply:
We just need this to execute prior to the compile phase, which should happen either sync or async. Given that we're not actively manipulating the compile params here, we might as well use async and whatever benefit that might get us.

Your example PR was helpful, thank you. I thought about going with a Promise.all rather than running them in series. Something like:

```js
    const foundFiles = await Promise.all(this.files.map(async (file) => {
      try {
        const exists = await fs.pathExists(file.absolute)

        return exists ? file : null
      } catch (e) {
        return null
      }
    }))

    this.files = foundFiles.filter((file) => file !== null)
```

Thoughts?

Reply:
Seems fine 💯